clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -413,12 +524,22 @@ public Map<TopicPartition, OffsetAndMetadata> committed(final Set<TopicPartition
     @Override
     public Map<TopicPartition, OffsetAndMetadata> committed(final Set<TopicPartition> partitions,
                                                             final Duration timeout) {
-        maybeThrowInvalidGroupIdException();
-        if (partitions.isEmpty()) {
-            return new HashMap<>();
+        long start = time.nanoseconds();
+        try {
+            maybeThrowInvalidGroupIdException();
+            final Map<TopicPartition, OffsetAndMetadata> offsets;
+            offsets = eventHandler.addAndGet(new OffsetFetchApplicationEvent(partitions), time.timer(timeout));
+            if (offsets == null) {
+                throw new TimeoutException("Timeout of " + timeout.toMillis() + "ms expired before the last " +
+                        "committed offset for partitions " + partitions + " could be determined. Try tuning default.api.timeout.ms " +
+                        "larger to relax the threshold.");
+            } else {
```

Comment:
the else is not needed

Reply:
@philipnee I added the `else` block as that's what appears in the `KafkaConsumer` implementation of `committed()`. I didn't see anything analogous as I looked into the `OffsetFetchApplicationEvent` path. Is it done somewhere else? Or do we not need it anymore?